web/public/app.dev.js

```diff
@@ -64,21 +67,34 @@ class App extends React.PureComponent {
       .then(res => {
         self.appVersion = res.headers.get('Version')
       });
-
+    
     this.setupWebSocket();
-  }
+}
 
   setupWebSocket() {
     try {
       let uri = window.location.protocol === 'https:' ? 'wss:' : 'ws:';
       uri += '//' + window.location.host;
-      uri += window.location.pathname + 'ws';
+      uri += window.location.pathname + 'ws'
+      let host_proxy_list = []
       this.ws = new WebSocket(uri);
-
       this.ws.onmessage = (evt) => {
         // exit early if we get bad JSON data
         if (evt.data === '') return
         const wsData = JSON.parse(evt.data)
+
+        if (host_proxy_list.includes(wsData['HostProxy']) === false) {
+          host_proxy_list.push(wsData['HostProxy'])
+        }
+
+        var result = host_proxy_list.map(e => ({id: e, name: e}))
+
+        this.setState({host_list: result});
```

Comment:
this may be ok... but this is going to fire ALOT, so if you run into performance issues (react working too hard) then you may need to throttle this method to only happen every 200ms or something like that.